### ISSUE TITLE
[C+ Checklist Needs Completion] [QBO Export] [$250] iOS mWeb: External links do not work from popover menus

### ISSUE THREAD

**Issue by arosiclair** [Expensify/App#45417]
**Version Number:** N/A
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** N/A
**Logs:** N/A
**Expensify/Expensify Issue URL:**
**Issue reported by:** @kosmydel 
**Slack conversation:** https://expensify.slack.com/archives/C036QM0SLJK/p1721049986353069?thread_ts=1720453976.209939&cid=C036QM0SLJK

## Action Performed:

1. On iOS mWeb, log in as a policy admin with a Xero/QBO connection
2. Avatar > Workspaces > {policy} > Connections > 3 dot menu > Sync Now

## Expected Result:
A window is opened that links to the integration

## Actual Result:
Nothing happens and the link doesn't work

## Workaround:
N/A

## Platforms:
Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/30ad3ddf-2575-46ce-90fc-b031887bad6d

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d607e7cf649f3935</li>
        <li>Upwork Job ID: 1812925016452625460</li>
        <li>Last Price Increase: 2024-07-15</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 103137581</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sonialiap</details>

**Comment by arosiclair** [comment]
Info from @kosmydel:

> Hey, during the development of integrations reconnection, I encountered a problem with links to external pages called from the three-dot menu. On the mWeb Safari it is impossible to open external links from the PopoverMenu (ThreeDotsMenu). After some investigation, the root cause of it is that [we call the `onSelected` function from `onModalHide`](https://github.com/Expensify/App/blob/951e3fe966870e7abbe86ff810659925eb6d2434/src/components/PopoverMenu.tsx#L182). The iOS safari, to open a new window, [requires us to do that in the `onClick` event](https://stackoverflow.com/questions/59792016/safari-blocking-popups/59792088#59792088). That change would require more investigation, and it would be nice to do that in a separate PR, as it might break some things. Could we create a new GH issue for that (someone from SWM will be able to handle it), so I can proceed with my PR. [Here is a minimal reproducible example.](https://github.com/Expensify/App/compare/main...software-mansion-labs:expensify-app-fork:%40kosmydel/popover-links-not-working)

**Comment by arosiclair** [comment]
@war-in can you comment so I can assign you? Thanks

**Comment by kabeer95** [comment]
# [QBO Export] [$250] iOS mWeb: External links do not work from popover menus.mb
## Problem Statement
External links do not work when accessed from popover menus in the iOS mWeb version of QBO Export.

## Root Cause
The current implementation of external links in popover menus does not account for the specific requirements of the iOS mWeb platform, resulting in links not functioning as expected.

### Solution
To fix this issue, we can update the implementation of external links in popover menus by adding the following code:

javascript
```
// Update external link implementation
document.addEventListener('click', (e) => {
  if (e.target.tagName === 'A' && e.target.href.startsWith('http')) {
    e.preventDefault();
    window.open(e.target.href, '_system');
  }
});
```


### Benefits
Improved User Experience: Fixing external links in popover menus ensures users can access relevant information without interruption.
Increased User Adoption: By providing a seamless experience, we aim to enhance user satisfaction and encourage higher adoption rates.

**Comment by arosiclair** [comment]
@kabeer95 please do not post proposals on issue that do not have `Help Wanted`

**Comment by war-in** [comment]
> @war-in can you comment so I can assign you? Thanks

@arosiclair @zfurtak is going to take care of this issue :) Please assign her here 🙏

**Comment by zfurtak** [comment]
Hello there! I've been looking into this recently and I've made a few discoveries 👩🏽‍💻
Currently we're not able to open an external link, when clicking on items in the `PopoverMenu` because Safari browser requires a direct interaction from the user to open a popover. This is because the function is called in `onHideModal`. 
However, when I tried to move this call to the `onPress` function some strange things started to happen. For example on iOS some modals open with height: 0 and width: 0 because the `PopoverMenu` doesn't have enough time to close, also some of the buttons don't respond. 
I've been looking through all the usages of `PopoverMenu` in the codebase and testing different approaches, but so far I haven't found the right, universal one. 
I'm still searching 🕵️‍♀️

**Comment by arosiclair** [comment]
I think https://github.com/Expensify/App/issues/46852 is the only regression that still needs to be addressed after these changes.

**Comment by sonialiap** [comment]
@arosiclair sounds like this is not ready for payment and we're waiting on https://github.com/Expensify/App/issues/46852, is that right?

**Comment by arosiclair** [comment]
I don't think we need to block payment on that issue. Just account for the regressions that were caught from this one.

**Comment by sonialiap** [comment]
Payment summary:
@shubham1206agra $125 due to regression - orig offer expired, sent [new offer](https://www.upwork.com/ab/applicants/1824047744623949643/job-details) in upwork - please complete the [checklist](https://github.com/Expensify/App/issues/45417#issuecomment-2270330080) and accept offer

**Comment by sonialiap** [comment]
@shubham1206agra paid, please complete the [checklist](https://github.com/Expensify/App/issues/45417#issuecomment-2270330080) ;)

**Comment by sonialiap** [comment]
@greg-schroeder I'm OOO Aug 19-30, adding leave buddy. 
Status: waiting for @shubham1206agra to complete the checklist, then submit regression test to QA

**Comment by greg-schroeder** [comment]
bump @shubham1206agra for checklist

**Comment by greg-schroeder** [comment]
Filed regression test, closing
